cmd/dep/status.go

```diff
@@ -793,12 +793,23 @@ func collectConstraints(ctx *dep.Ctx, p *dep.Project, sm gps.SourceManager) (con
 			// Get project constraints.
 			pc := manifest.DependencyConstraints()
 
+			imports, err := projectImports(sm, proj)
```

Comment:
If I understand this correctly, `imports` contains all the imports of `proj` right? Not the root project?
Applicable/working constraints are in respect to the root project, `p`. That's why [`getApplicableConstraints()`](https://github.com/golang/dep/blob/7ccbfed88623a8565f300d02fe76c0c3a1daaac8/gps/rootdata.go#L115) uses `rootdata` to walk through all the import paths.

Correct me if I'm not seeing it the wrong way 🙂

Reply:
Opps! My bad, I misunderstood. This makes it simpler though! 

I also looked the issue you referred me to, and it does look like a function could be written to satisfy both issues. Which file would the best placement be for it?